src/usr/local/www/diag_reboot.php

```diff
@@ -50,11 +50,26 @@
 	if (DEBUG) {
 		print_info_box(gettext("Not actually rebooting (DEBUG is set true)."), 'success');
 	} else {
-		print('<div><pre>');
-		system_reboot();
-		print('</pre></div>');
+		switch ($_POST['rebootmode']) {
+			case 'FSCKReboot':
+				mwexec('/sbin/nextboot -e "pfsense.fsck.force=5"');
+				print('<div><pre>');
```

Comment:
This is still repeated too much. It's used in every case. Put the opening `print` before the `switch` and the closing `print` after the `switch`.

Reply:
fixed